descriptor.go

```diff
@@ -1295,6 +1296,17 @@ func parseDescriptors(i *astikit.BytesIterator) (o []*Descriptor, err error) {
 				// previously therefore we must fetch bytes in descriptor functions and seek at the end
 				offsetDescriptorEnd := i.Offset() + int(d.Length)
 
+				// <Hack>: assign the original bytes to an internal byte slice for use when reserialising later
+				// TODO fix this to actually serialise the struct
+				origOffset := i.Offset()
+				if d.originalBytes, err = i.NextBytes(int(d.Length)); err != nil {
+					err = fmt.Errorf("astits: fetching original bytes failed: %w", err)
+					return
+				}
+				// Reset iterator so parsing can continue
+				i.Seek(origOffset)
+				// </Hack>
```

Comment:
Sorry, could you please explain why this is necessary? I don't have the context to understand what's going on here.

Reply:
The correct way to do this would be to look at the internal values of the struct and create a byte sequence. That way if you change the values of the struct, it will be reflected when serialising. There are 45 Descriptor types currently supported in this library, each with their own byte sequence. I also don't anticipate us having to edit any Descriptors, so for now we just pull the original bytes into a slice that is attached to the Descriptor struct. That way when we re-serialise it we just copy rather than reserialising directly.

Reply:
</Hack>

Reply:
Okay, SGTM. Thanks for clarifying 🙂